### ISSUE TITLE
User-defined types not being recognized

### ISSUE THREAD

**Issue by drkameleon** [drkameleon/iswift#42]
Using any user-defined type, either using the `@interface` declaration or the `#import`ed header should automatically be registered as a type.

And thus avoid the resulting `unrecognized #identifier` errors.

**Comment by bre7** [comment]
Error using enums (they are defined outside @interface). Example:

```objectivec
typedef NS_ENUM (NSInteger, MyEnum) {
    MyEnumCase1,
    MyEnumCase2
};

@interface SomeClass
+ (MyEnum) ....
```

**Comment by drkameleon** [comment]
Hmm ... That's weird...

a) has it worked for you inside interface?
b) what if you try it at the online version? http://iswift.org/try

**Comment by drkameleon** [comment]
Issue merged here (to keep things clean :) ): https://github.com/drkameleon/iswift/issues/50

**Comment by wahlstedt** [comment]
Are typedefs supposed to work in 1.5? I have typedefs in an #import but it gives "Unexpected: #identifier" errors.


For example, `ulong32` is not recognised here:
```
#import "typedef.h"

static const ulong32 INIT_PA[BF_SUBKEYS] = {
        0x243F6A88L, 0x85A308D3L, 0x13198A2EL, 0x03707344L,
        0xA4093822L, 0x299F31D0L, 0x082EFA98L, 0xEC4E6C89L,
        0x452821E6L, 0x38D01377L, 0xBE5466CFL, 0x34E90C6CL,
        0xC0AC29B7L, 0xC97C50DDL, 0x3F84D5B5L, 0xB5470917L,
        0x9216D5D9L, 0x8979FB1BL
};
```

In typedef.h:
```
typedef signed char		schar;
typedef unsigned char	uchar;

typedef signed short	sshort;
typedef unsigned short	ushort;

typedef signed int		slong32;
typedef unsigned int	ulong32;

typedef signed long long slong64;
typedef unsigned long long ulong64;
```

**Comment by drkameleon** [comment]
> Are typedefs supposed to work in 1.5? I have typedefs in an #import but it gives "Unexpected: #identifier" errors.

I'm looking into it.

Re: `#import "typedef.h"`, you mean you expected the app to parse the contents of the imported header as well? (well, in theory, it should. I don't disagree with that. But, for now, it just knows only what it sees - without actually processing pre-processor statements)

**Comment by wahlstedt** [comment]
I did try to move in the typedefs into the .m file but got the same error. But yes, I thought it resolved imports. Hope that can be added in a future update! :)

**Comment by drkameleon** [comment]
> But yes, I thought it resolved imports. Hope that can be added in a future update! :)

Well, to be honest, I've thought of that too. But right now my.... flightplan is:

- Make it correctly parse all (or at least a 99%) of all valid sources
- Further optimizations to the output
- Multiple sources (+ full Xcode project) conversion - which, will include things like resolving the `#import` directives

> I did try to move in the typedefs into the .m file but got the same error.

This is what I'd suggest - for now that is. Putting everything in the very same source. Now, the thing that it still throws an error simply means that there really *is* something wrong. And obviously not with your code. `typedef enum`s and `typedef NS_ENUM`s and the like were first implemented in the very last update (1.5). Obviously I missed the rest of the `typedef`s - but, now that you mentioned it, it's in my to-do list. :)

**Comment by drkameleon** [comment]
Moving discussion to: https://github.com/drkameleon/iswift/issues/52 (to keep things... clean :-) )
